### PR TITLE
Enhancement(CX-1967): Replace my collection lab feature with Echo Flag

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **Enhancement**

This PR resolves [CX-1967]

### Description
Replaces use of lab features for my collection with Echo flag

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Replace my collection lab feature with Echo Flag - kizito

</details>

[CX-1967]: https://artsyproduct.atlassian.net/browse/CX-1967

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Replace my collection lab feature with Echo Flag - kizito)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d37ad91caa003bb8740215e8d6447160d16f7fd9
</p>

**Review by AlicanAkyuz — APPROVED** [review]
looking good!

File: src/lib/Scenes/MyProfile/__tests__/MyProfileSettings-tests.tsx [hunk 1]
```diff
@@ -1,48 +1,16 @@
...
-  const getWrapper = (mockResolvers = {}) => {
-    const tree = renderWithWrappers(<TestRenderer />)
-    act(() => {
-      mockEnvironment.mock.resolveMostRecentOperation((operation) =>
-        MockPayloadGenerator.generate(operation, mockResolvers)
-      )
-    })
+describe(MyProfileSettings, () => {
+  const getWrapper = () => {
+    const tree = renderWithWrappers(<MyProfileSettings />)
```
Comment (MounirDhahri) [inline src/lib/Scenes/MyProfile/__tests__/MyProfileSettings-tests.tsx:13]: maybe we can test which variation of the home screen here is visible by platform depending on the feature flag

**Review by MounirDhahri — COMMENTED** [review]
Just a small comment and we should be clear to release this

File: src/lib/Scenes/Favorites/Favorites.tsx [hunk 2]
```diff
@@ -58,34 +54,42 @@ export const Favorites: React.FC<Props> = ({
...
+    finalInitialTab = initialTab
+  } else {
+    console.warn("Specified Initial Tab is not available in tabs array.")
   }
+
+  // Set initial Tab
+  tabs.forEach((tab) => {
+    tab.initial = tab.title === finalInitialTab
+  })
+
```
[the remark this answers is not in the document]
Comment (lordkiz) [reply src/lib/Scenes/Favorites/Favorites.tsx:92]: Yesss and Thanks @AlicanAkyuz . The problem is that if we do these tab operations in an effect we will always have initialTab index set at 0 in `StickyTabPage` therefore making the initialTab prop useless. This is obviously stemming from here: https://github.com/artsy/eigen/blob/master/src/lib/Components/StickyTabPage/StickyTabPage.tsx#L48

And probably why we have always kept all these outside of effect. Eg: https://github.com/artsy/eigen/blob/master/src/lib/Scenes/Artist/Artist.tsx#L104

This is one bug I can't wrap my head around. 🤒

File: src/lib/Scenes/MyProfile/__tests__/MyProfileSettings-tests.tsx (hunk 1 above)
Comment (lordkiz) [reply src/lib/Scenes/MyProfile/__tests__/MyProfileSettings-tests.tsx:13]: @MounirDhahri I have decoupled MyProfile from MyProfileSettings. Users will never see MyProfileSettings without the feature flag on. So the feature flag has no effect here. I probably should add that check instead in MyProfile-tests.

Comment (MounirDhahri) [reply src/lib/Scenes/MyProfile/__tests__/MyProfileSettings-tests.tsx:13]: cool then. it's your decision then, feel free to merge then

File: src/lib/Scenes/MyCollection/MyCollection.tsx [hunk 3]
```diff
@@ -27,6 +28,19 @@ export function refreshMyCollection() {
   RefreshEvents.emit(REFRESH_KEY)
 }
 
+const featureFlagKey = Platform.select({
+  android: "AREnableMyCollectionAndroid",
+  default: "AREnableMyCollectionIos",
+}) as "AREnableMyCollectionIos" | "AREnableMyCollectionAndroid"
```
Comment (brainbicycle) [inline src/lib/Scenes/MyCollection/MyCollection.tsx:34]: good thinking 🙌 had forgotten we would need two flags

**Review by brainbicycle — APPROVED** [review]

**Pull request merged**